In [1]:
#Import necessary libraries
import pandas as pd
import numpy as np
from scipy.stats import ttest_ind

In [2]:
data = pd.read_csv ("dataset w7.csv")
print(data)

    Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female      

In [3]:
#Check for Null values
data.isnull().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

### 1. Carry out descriptive analytics on the dataset.

In [4]:
data['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

In [5]:
data[('Total Marks')].describe()

count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000
Name: Total Marks, dtype: float64

In [6]:
data[('Test preparation')].describe()

count       28
unique       2
top       none
freq        21
Name: Test preparation, dtype: object

In [7]:
data.groupby('Gender')['Test preparation'].value_counts()

Gender  Test preparation
female  none                12
        completed            4
male    none                 9
        completed            3
Name: Test preparation, dtype: int64

In [8]:
data.groupby('Gender')['Total Marks'].mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [9]:
data.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

#### Inferences

1. There are 28 samples and no null values.
2. 12 male and 16 female
3. The mean, minimum and maximum score of the total marks are 32.321429, 12 and 50 respectively
4. 12 females and 9 males didn't attend the test preparation whereas 4 females and 3 males
5. The average total marks for females(33.75) are greater than males(30.41)
6. The average total marks for those who completed test preparation(44.57) are greater than for those who didn't attend the test preparation(28.23)


### 2. Check whether female and male students scored the same marks.

   
H0: Female and Male students scored same marks.

Ha: Female and Male students doesn't score same marks.

In [10]:
maleMarks = data[data["Gender"] == "male"][["Total Marks"]]
femaleMarks = data[data["Gender"] == "female"][["Total Marks"]]

t_score,p_value = ttest_ind(maleMarks,femaleMarks)
print('t-score:', t_score, ' p-value:', p_value)

t-score: [-0.69408853]  p-value: [0.49378223]


#### Inference

As p-value (0.49378223) is greater than alpha (0.05), we failed to reject null hypothesis(H0) and reject alternate hypothesis(Ha).

As per the hypothesis testing, we can conclude that there is not much significant difference in marks based on gender

### 3. Whether test preparation helps the students?


H0: There is no significant difference between the scores of students who completed test preparation than who didn't attend the test preparation.

Ha: There is a significant difference between the scores of students who completed test preparation than who didn't attend the test preparation.

In [11]:
completedMarks = data[data["Test preparation"]=="completed"][["Total Marks"]]
noneMarks = data[data['Test preparation']=="none"][["Total Marks"]]

t_score_1,p_value_1 = ttest_ind(completedMarks,noneMarks)
print('t-score:', t_score_1, ' p-value:', p_value_1)


t-score: [3.61436375]  p-value: [0.00126713]


#### Inference

As p-value(0.00126713) is less than alpha(0.05), we reject the null hypothesis(H0) and failed to reject the alternate hypothesis(Ha)

Students who have completed test preparations scored higher than those who didn't attend the test preparations. Therefore, we conclude that the test preparation significantly helped the students.